### ISSUE TITLE
Rogue notifications on v3.4 with OSX Safari client on Centos 7 server 

### ISSUE THREAD

**Issue by giankdm** [silverstripe/silverstripe-framework#5772]
I've got a similar problem to #5384 on v3.4. 

In the dev env (Ubuntu Trusty 14) the issue doesn't manifest, however in the stage env (CentOS 7) on the destination server I've got a slew of notifications. It happens in OSX Safari, doesn't happen in OSX Google Chrome. I think a new browser update added to the problem.
![1f854960-3c88-11e6-8d4a-59af103408c4](https://cloud.githubusercontent.com/assets/3578378/16564608/b60651e2-4208-11e6-8ce2-ac974bd140ce.jpg)

**Comment by tractorcow** [comment]
Can you tell me what X-Status header your responses are sending?

**Comment by giankdm** [comment]
Site is served over SSL. Status of the HTTPS XHR requests is `HTTP/2.0 200`.
I tried disabling SSL: status of the XHR requests becomes `OK`.

So I updated the  `ignoredMessages` array in `framework/admin/javascript/LeftAndMain.js`, added `HTTP/2.0 200` to the list and the popup did not appear.

`ignoredMessages = ['OK', 'success', 'HTTP/2.0 200'];`

**Comment by dhensby** [comment]
@giankdm what software are you using to get HTTP/2 on CentOS 7?

**Comment by giankdm** [comment]
It is a Plesk 12.5 installation over Centos 7, with Apache and nginx managed by Odin, latest updates applied. So it is pretty common. When you install a certificate, apparently it triggers the new HTTP 2.0/SPDY protocol on certain browsers, and it appears to have some slightly different statuses.

**Comment by dhensby** [comment]
HTTP2 doesn't send http status strings, just codes. which is the "problem" - I think this is a bug with jquery / js / safari not with the framework - but we do need to decide if we want to try to mitigate it

**Comment by giankdm** [comment]
Apparently Safari translates the `200` code into a string with `HTTP/2.0 ` attached. However, the same environment in Google Chrome reports a different revenue: the status is reported to be `200`. I think it should be mitigated (Safari has a longer update cycle than Chrome, and if no tests are failed, it is pretty simple to circumvent). Meanwhile we could try to create a minimum code sample to reproduce the bug and submit an Apple Radar and/or a jQuery issue.

**Comment by giankdm** [comment]
Reported on Safari Radar with issue #27171810. Please also add to the discussion there.

**Comment by dhensby** [comment]
This issue was reported to webkit quite some time ago, I believe.

I think the long term solution is to not rely on the `statusText` because it looks like it's defunct in HTTP/2

**Comment by giankdm** [comment]
you mean using only the code result of `xhr.status`?

**Comment by dhensby** [comment]
I think we have to use a custom header or even a dedicated part of the API calls to force a message popup rather than relying on non-standard status text

**Comment by giankdm** [comment]
I understand. If we meanwhile update the `ignoredMessages` array, how to be sure notifications still get shown as usual in all browsers?

**Comment by dhensby** [comment]
as a short term fix, I'd accept a PR to the effect of adding `HTTP/2.0 200` to the list of `ignoredMessages`.

This shouldn't affect the notifications being shown in other browsers

**Comment by tractorcow** [comment]
Can someone able to reproduce this issue please PR the above fix suggested by @dhensby ?

It would be to edit this line in LeftAndMain.js from:

```js
ignoredMessages = ['OK', 'success'];
```

to

```js
ignoredMessages = ['OK', 'success', 'HTTP/2.0 200'];
```

**Comment by giankdm** [comment]
I want to do it, but it would be my first contribution on github; I need a day or two to make sure to do this right.

**Comment by giankdm** [comment]
Sorry to hijack the discussion with a different topic, but I don't really know where to start. I've done this steps to contribute

```
mkdir silverstripe/silverstripe-framework
cd silverstripe/silverstripe-framework/
git clone https://github.com/silverstripe/silverstripe-framework.git .
git pull origin master
git checkout -b pr-issue-5772
vi ./admin/client/dist/js/LeftAndMain.js
git add admin/client/dist/js/LeftAndMain.js
git commit -m 'Quick fix for HTTP/2.0 on OSX/Safari v9.1.1'
git push origin pr-issue-5772
```

but I received this error

```
git push origin pr-issue-5772
Username for 'https://github.com': giankdm
Password for 'https://giankdm@github.com': ********
remote: Permission to silverstripe/silverstripe-framework.git denied to giankdm.
fatal: unable to access 'https://github.com/silverstripe/silverstripe-framework.git/': The requested URL returned error: 403
```

Sorry again but I thought it would be better to ask for help than let this conversation die.

Who can point me to the right procedure to contribute to a repository you don't have direct write access permission to?

**Comment by kinglozzer** [comment]
@giankdm You have to “fork” the repository (near the top-right), which will create a copy of the repository on your account. Then run:

```
git remote rename origin upstream
git remote add origin https://github.com/giankdm/silverstripe-framework
git push origin pr-issue-5772
```

This will rename https://github.com/silverstripe/silverstripe-framework to “upstream”, and add your fork as a second remote called “origin”. Once you’ve pushed your branch to _your fork_, you should then see an option to create a pull request in Github’s UI.

Good luck! :)

**Comment by muskie9** [comment]
[Here are the docs for contributing too](https://docs.silverstripe.org/en/4.0/contributing/code/)

**Comment by giankdm** [comment]
Thank you to everyone. The docs say

> If you are modifying CSS or JavaScript files in core modules, you'll need to regenerate some files. Please check out our client-side build tooling guide for details.

is this notice relevant for v3.4 or just for v4.x?

**Comment by muskie9** [comment]
I'll defer to @kinglozzer for that one as I've never worked with the css/js files.

**Comment by kinglozzer** [comment]
@giankdm That applies to 4.x, but I think this would be best targeted at 3.1 (as that’s what #5384 was merged into). So the steps should be something like:

```
git checkout 3.1
git checkout -b 5772-notifications
vi ./admin/javascript/LeftAndMain.js
git add admin/javascript/LeftAndMain.js
git commit -m 'Quick fix for HTTP/2.0 on OSX/Safari v9.1.1'
git push origin pr-issue-5772
```

**Comment by dhensby** [comment]
Now fixed. Thanks so much @giankdm
